test/conftest.py

```diff
@@ -298,6 +301,64 @@ def fixture_standby2_controller(session_tmpdir, mysql_standby2, default_backup_s
         controller.stop()
 
 
+@pytest.fixture(scope="function", name="empty_controller")
+def fixture_empty_controller(
+    session_tmpdir, mysql_empty: MySQLConfig, default_backup_site: BackupSiteInfo
+) -> Iterator[tuple[Controller, MySQLConfig]]:
+    controller = build_controller(
+        Controller,
+        default_backup_site=default_backup_site,
+        mysql_config=mysql_empty,
+        session_tmpdir=session_tmpdir,
+    )
+    try:
+        yield controller, mysql_empty
+    finally:
+        controller.stop()
+
+
+@pytest.fixture(scope="function", name="empty_controller_in_small_disk")
+def fixture_empty_controller_in_small_disk(
+    session_tmpdir, mysql_empty: MySQLConfig, default_backup_site: BackupSiteInfo
+) -> Iterator[tuple[Controller, MySQLConfig]]:
+    sub_dir = Path(session_tmpdir().strpath) / "small_disk"
+    with mount_tmpfs(path=sub_dir, size=8) as tmpfs_dir:
+        controller = build_controller(
+            Controller,
+            default_backup_site=default_backup_site,
+            mysql_config=mysql_empty,
+            session_tmpdir=lambda: LocalPath(tmpfs_dir),
+        )
+        try:
+            yield controller, mysql_empty
+        finally:
+            controller.stop()
+
+
+@contextmanager
+def mount_tmpfs(path: Path, size: int) -> Iterator[Path]:
+    """Mount a tmpfs filesystem at the given path and unmount it when done.
+
+    Args:
+        path: The path to mount the tmpfs filesystem at (will create a subdirectory there).
+        size: The size of the tmpfs filesystem in megabytes.
+
+    Yields:
+        The path the tmpfs filesystem was mounted at.
+    """
+    sub_dir = path / random_basic_string(20, prefix="small_disk_")
+    try:
+        sub_dir.mkdir(parents=True, exist_ok=True)
+        subprocess.check_call(["sudo", "mount", "-t", "tmpfs", "-o", f"size={size}m", "tmpfs", str(sub_dir)])
```

Comment:
Tests requiring `sudo` are probably not the best idea, but we can leave it in for now as it runs on github. But I would probably split the usage of this into a separate integration test to make sure we can still run the bulk of the tests locally without any special treatment. This way we could also split out the fixtures to this special file, as they are used only from one file.

Reply:
Agreed. It's @rikonen 's suggestion, which works great for what we need. It's definitely not ideal but shouldn't be a blocker for now :-).